db/db_block_cache_test.cc

```diff
@@ -305,6 +305,37 @@ TEST_F(DBBlockCacheTest, IndexAndFilterBlocksOfNewTableAddedToCache) {
             TestGetTickerCount(options, BLOCK_CACHE_INDEX_HIT));
 }
 
+// With fill_cache = false, fills up the cache, then iterates over the entire
+// db, verify dummy entries inserted in `BlockBasedTable::NewDataBlockIterator`
+// does not cause heap-use-after-free errors
```

Comment:
Would be more clear if we explicitly say that the test targets COMPILE_WITH_ASAN runs.